src/background.js

```diff
@@ -0,0 +1,38 @@
+function runShowMeTheCode(tabId, tab) {
+    function isGitHub(tab) {
+        return urlMatches(tab, /https:\/\/github\.com\/.*/)
+    };
+
+    function isBlobView(tab) {
+        return urlMatches(tab, /.*\/blob\/.*/);
+    };
+
+    function pullRequestFileView(tab) {
+        return urlMatches(tab, /.*\/pull\/.*\/files/);
+    };
+
+    function urlMatches(tab, expression) {
+        return tab.url && tab.url.match(expression);
```

Comment:
This check is needed for chrome configuration page
18 gru 2015 6:14 PM "Łukasz Nowak" <notifications@github.com> napisał(a):

> In src/background.js
> <https://github.com/Viridovix/show-me-the-code-github/pull/32#discussion_r48047276>
> :
>
> > @@ -0,0 +1,38 @@
> > +function runShowMeTheCode(tabId, tab) {
> > +    function isGitHub(tab) {
> > +        return urlMatches(tab, /https:\/\/github\.com\/.*/)
> > +    };
> > +
> > +    function isBlobView(tab) {
> > +        return urlMatches(tab, /.*\/blob\/.*/);
> > +    };
> > +
> > +    function pullRequestFileView(tab) {
> > +        return urlMatches(tab, /.*\/pull\/.*\/files/);
> > +    };
> > +
> > +    function urlMatches(tab, expression) {
> > +        return tab.url && tab.url.match(expression);
>
> I get it - but all pages do have url anyway, right? It's the url content
> matching proper expresion what is interesting for us.
> Question is - do we need tab.url or tab.url.match(expression) is
> sufficient?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Viridovix/show-me-the-code-github/pull/32/files#r48047276>
> .
>